### PR TITLE
feat(form): validate() return promise

### PR DESCRIPTION
**Pull request by ibufu**
#6568 
* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by yuchonghua** [comment]
Can [upgrade](https://github.com/ElemeFE/element/pull/6529) at the same time [async-validator](https://www.npmjs.com/package/async-validator) version 1.8.1, because in the async-validator support in the [1.8.0](https://github.com/yiminghe/async-validator/commit/c26eb70150723d5a7f8dff6c99808ec77397e7d5) [promise validation](https://github.com/yiminghe/async-validator/pull/59), very grateful

/ping @ibufu @baiyaaaaa @Leopoldthecoder

File: packages/form/src/form.vue [hunk 1]
```diff
@@ -71,6 +71,9 @@
         // 如果需要验证的fields为空，调用验证时立刻返回callback
         if (this.fields.length === 0 && callback) {
           callback(true);
```
Comment (yuchonghua) [inline packages/form/src/form.vue]: 建议
```js
callback && callback(true)

```

File: packages/form/src/form.vue [hunk 2]
```diff
@@ -71,6 +71,9 @@
         // 如果需要验证的fields为空，调用验证时立刻返回callback
         if (this.fields.length === 0 && callback) {
```
Comment (yuchonghua) [inline packages/form/src/form.vue]: 个人见解，不确定是否需要这样修改，因为我认为，可能没有传入回调，但是是一个Promise
```js
this.fields.length === 0
```

Comment (yuchonghua) [inline packages/form/src/form.vue]: 个人见解，不确定是否需要这样修改，因为我认为，可能没有传入回调，但是是一个Promise
```js
this.fields.length === 0
```

**Comment by Leopoldthecoder** [comment]
#7405

**Pull request closed without merging**